sys/picolibc_syscalls_default/syscalls.c

```diff
@@ -282,13 +330,23 @@ int fcntl (int fd, int cmd, int arg)
  */
 off_t lseek(int fd, _off_t off, int whence)
 {
-    int res = vfs_lseek(fd, off, whence);
+    if (fd < VFS_FD_OFFSET || !IS_USED(MODULE_VFS)) {
+        errno = ENOTSUP;
+        return -1;
+    }
+
+#ifdef MODULE_VFS
+    int res = vfs_lseek(fd - VFS_FD_OFFSET, off, whence);
     if (res < 0) {
         /* vfs returns negative error codes */
         errno = -res;
         return -1;
     }
     return res;
+#else
+    (void) off;
+    (void) whence;
```

Comment:
```suggestion
    (void)off;
    (void)whence;
```

Just place those unconditionally at the start of the function; no need to guard them as they do no harm. More instances elsewhere.

Reply:
Well they *do* catch errors (if one actually compiles with VFS :sweat_smile:)